pkg/expr/func_arr.go

```diff
@@ -220,3 +222,81 @@ func slice(arr interface{}, start, end int) interface{} {
 
 	return v.Slice(start, end).Interface()
 }
+
+// sortSlice sorts slice
+func sortSlice(arr interface{}, desc bool) (out interface{}, err error) {
+	if arr, err = toSlice(arr); err != nil {
+		return
+	}
+
+	// sort slice of native type values
+	var (
+		v  = reflect.ValueOf(arr)
+		vi = v.Interface()
+	)
+
+	// sort the array if error is nil;
+	// we trap the error(s) in the outer scope from the less function,
+	// and returning the last error at a time since all error will the same in almost every scenario.
```

Comment:
The error management is a bit off --

The `err` variable used in `c, err = casted.Compare(stv[j])`, `err = fmt.Errorf("cannot compare %s and %s: unknown state",...)`, and `err = fmt.Errorf("cannot compare %s and %s: unknown state", getKind(nVal), getKind(mVal))` is defined in the outer scope and is the same.

With that said, if (for example) `err = fmt.Errorf("cannot compare %s and %s: unknown state", getKind(nVal), getKind(mVal))` sets the error, the upper line (`c, err = casted.Compare(stv[j])`) can clear the error and make it seem like nothing happened.

What I would do is this:

```go
out, sortErr := sortSliceStable(vi, func(i, j int) bool {
	if err != nil {
		return false
	}

	var (
		nVal = v.Index(i)
		mVal = v.Index(j)
	)
	if desc {
		nVal = v.Index(j)
		mVal = v.Index(i)
	}

	switch getKind(vi) {
	// ...
	// ...
	// ...
	default:
		// sort slice of typed values
		if stv, is := arr.([]TypedValue); is {
			if casted, ok := stv[i].(Comparable); ok {
				var c int
				c, err = casted.Compare(stv[j])
				if err != nil {
					return false
				}
				// ...
			} else {
				err = fmt.Errorf("cannot compare %s and %s: unknown state", stv[i].Type(), stv[j].Type())
				return false
			}
		}
		err = fmt.Errorf("cannot compare %s and %s: unknown state", getKind(nVal), getKind(mVal))
		return false
	}
})
```

The above snippet adds a little guard statement to the very top to skip the entire processing in case something went wrong.